src/libraries/Common/src/Interop/Unix/System.Native/Interop.CopyFile.cs

```diff
@@ -9,7 +9,28 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
+        private static volatile int s_tryCopyFileRange;
+
+        private static bool TryCopyFileRange
+        {
+            get
+            {
+                int tryCfr = s_tryCopyFileRange;
+                if (tryCfr == 0) // Uninitialized.
+                {
+                    // Avoid known issues with copy_file_range that are fixed Linux 5.3+ (https://lwn.net/Articles/789527/).
+                    s_tryCopyFileRange = tryCfr =
+                        !OperatingSystem.IsOSPlatform("LINUX") ||
```

Comment:
This may have a problem on Android. Android has Linux kernel that may have a problem that this is trying to workaround, but it is not `IsOSPlatform("LINUX")`.

Reply:
I'll fix this when moving the check to native code.